### PR TITLE
Add documentation for DFU

### PR DESCRIPTION
**Pull request by radurentea**

### CODE REVIEW

File: docs/tutorial/dfu.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Device Firmware Upgrade via USB.
+
+## Requirements:
+
+- ESP32-S2 or an ESP32-S3 chip
+- You will need to do some electrical connection work. (Here is a guide for the S2 board: https://blog.espressif.com/dfu-using-the-native-usb-on-esp32-s2-for-flashing-the-firmware-b2c4af3335f1)
+- The ESP32-S2 chip needs to be in bootloader mode for the detection as a DFU device and flashing. This can be achieved by pressing the “reset” button, while holding the “boot” button pressed
```
Comment (brianignacio5) [inline docs/tutorial/dfu.md:7]: Also ESP32-S3 or just the chip

File: docs/tutorial/dfu.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+<p>
+  <img src="../../media/tutorials/dfu/select_device.png" alt="Select device" height="500">
+</p>
+
+2. Build the project
+<p>
+  <img src="../../media/tutorials/dfu/build_project.png" alt="Build Project" height="500">
+</p>
+
+3. Flash
```
Comment (brianignacio5) [inline docs/tutorial/dfu.md:23]: The user also needs to choose the Flash mode UART from the status bar or the `espIdf.selectFlashMethodAndFlash` command

**Review by brianignacio5 — CHANGES_REQUESTED** [review]

File: docs/tutorial/dfu.md (hunk 1 above)
Comment (radurentea) [reply docs/tutorial/dfu.md:7]: Thanks!

File: docs/tutorial/dfu.md (hunk 2 above)
Comment (radurentea) [reply docs/tutorial/dfu.md]: Fixed it, thanks!

**Pull request merged**